js/apps/system/_admin/aardvark/APP/react/src/views/query/queryResults/CancelQueryButton.tsx

```diff
@@ -12,12 +12,10 @@ export const CancelQueryButton = ({
 }) => {
   const { onRemoveResult } = useQueryContext();
   const onCancel = async () => {
-    const route = getApiRouteForCurrentDB();
     try {
-      await route.request({
-        method: "PUT",
-        path: `/job/${asyncJobId}/cancel`
-      });
+      if (asyncJobId) {
+        await getCurrentDB().job(asyncJobId).cancel();
+      }
```

Comment:
I think instead of adding that `if (asyncJobId)` block, we should fix the type. From the usage of `<CancelQueryButton>`, it looks like we can assume we will always get `asyncJobId: string`

And if not so, then we should probably hide the button itself. Otherwise, this is a no-op. Good it got caught in this change!